### PR TITLE
Replace junit with kotlin test

### PR DESCRIPTION
**Pull request by luigi617**
### Description

Replaces JUnit-specific APIs with kotlin.test and cross-platform utilities in `runtime:testing`, as part of the effort to standardize on `kotlin-test` across our repositories.

#### New cross-platform utilities in runtime:testing

- **parameterized()** — cross-platform alternative to JUnit's `@ParameterizedTest`. Runs a test block for each case, collects all failures, and reports them at the end with case-level detail.
- **withTempDir()** — cross-platform alternative to JUnit's `@TempDir`. Creates a temporary directory using `kotlinx-io`, runs a block, and cleans up based on `TempDirCleanupMode` (ALWAYS, ON_SUCCESS, NEVER). Default is ON_SUCCESS (matches JUnit's `CleanupMode.ON_SUCCESS`).
- Added `kotlinx-io-core` dependency for cross-platform filesystem operations.

#### JUnit removals

- **@ParameterizedTest / @MethodSource** — Tests converted to use `parameterized()`.
- **@ParameterizedTest / @MethodSource / assumeTrue** — Provider-specific tests (canonical request, signature, string-to-sign) moved from the base class into `DefaultSigningSuiteTest` directly, eliminating the need for assumeTrue to skip tests when providers are null in `CrtSigningSuiteTest`.
- **@RegisterExtension** — Replaced JUnit's OpenTelemetryExtension with manual InMemoryMetricReader + SdkMeterProvider + OpenTelemetrySdk setup. Metrics are reset via `@BeforeTest`.
- **@EnabledForJreRange** — Removed from EcdsaJVMTest. Replaced with a runtime check that returns early if `SHA256withECDSAinP1363Format` is unavailable (Java 8). Also fixed the comment: this algorithm was 
introduced in Java 9, not Java 11.
- **@TestInstance(PER_CLASS)** — Removed from DokkaSmithyTest (unnecessary, no shared state).

#### Kept as-is (documented)

- **@EnabledIfSystemProperty** in ProxyTest.kt — Required for class-level skip of Docker-dependent proxy tests. Gradle sets the property based on environment (not CodeBuild + Linux). Replacing would require Gradle test filtering or runtime guards in @BeforeAll.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: runtime/runtime-core/jvm/test/aws/smithy/kotlin/runtime/hashing/EcdsaJVMTest.kt [hunk 1]
```diff
@@ -101,10 +99,11 @@ class EcdsaJVMTest {
         assertTrue(verifier.verify(signature))
     }
 
+    // SHA256withECDSAinP1363Format algorithm was introduced in Java 9, skip on Java 8
     @Test
-    // SHA256withECDSAinP1363Format algorithm was introduced in Java 11, skip on Java 8
-    @EnabledForJreRange(min = JRE.JAVA_11)
     fun testVerifyRawSignature() {
+        if (runCatching { Signature.getInstance("SHA256withECDSAinP1363Format") }.isFailure) return
```
Comment (ianbotsf) [inline runtime/runtime-core/jvm/test/aws/smithy/kotlin/runtime/hashing/EcdsaJVMTest.kt]: **Nit**: There's no reason to get the signature instance twice. You should capture it here so it can be used later:

```kotlin
val verifier = runCatching { Signature.getInstance("SHA256withECDSAinP1363Format") }.getOrNull() ?: return
```

File: runtime/testing/common/src/aws/smithy/kotlin/runtime/testing/Parameterized.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ *     listOf(
+ *         Triple("String", "\"foo\"", "\"foo\""),
+ *         Triple("Boolean", "false", "false"),
+ *     ),
+ * ) { (type, modelDefault, expectedDefault) ->
+ *     // ...
+ * }
+ * ```
+ */
+public fun <T> parameterized(cases: List<T>, test: (T) -> Unit) {
```
Comment (ianbotsf) [inline runtime/testing/common/src/aws/smithy/kotlin/runtime/testing/Parameterized.kt:48]: **Comment**: Very nice KDocs!

**Review by ianbotsf — APPROVED** [review]
Minor feedback only. Fix and ship!
